src/options/file_name.rs

```diff
@@ -28,12 +28,19 @@ impl Options {
 
 impl Classify {
     fn deduce(matches: &MatchedFlags<'_>) -> Result<Self, OptionsError> {
-        let flagged = matches.has(&flags::CLASSIFY)?;
+        let mode_opt = matches.get(&flags::CLASSIFY)?;
 
-        if flagged {
-            Ok(Self::AddFileIndicators)
-        } else {
-            Ok(Self::JustFilenames)
+        match mode_opt {
+            Some(word) => match word.to_str() {
+                Some("always") => Ok(Self::AddFileIndicators),
+                Some("auto" | "automatic") => Ok(Self::AutomaticAddFileIndicators),
+                Some("never") => Ok(Self::JustFilenames),
+                // Bare flag given, always add file indicators for backwards compatibility
+                None => Ok(Self::AddFileIndicators),
```

Comment:
as you put a base value in flag.rs this case is a parsing error more than anything.

Reply:
ack